src/Components/AddItem.jsx

```diff
@@ -1,26 +1,63 @@
 import React from 'react';
-import { useForm } from 'react-hook-form';
-import firebase from 'firebase';
 import { db } from '../lib/firebase';
+import { useForm } from 'react-hook-form';
+import { formatString } from '../lib/helpers.js';
 
 export default function AddItem({ token }) {
-  const { register, handleSubmit, reset, errors } = useForm();
-  const onSubmit = (data) => {
-    const currentList = db.collection('lists').doc(token);
+  const { register, handleSubmit, reset, setError, errors } = useForm();
+
+  const checkForDuplicateItem = async (currentList, itemName) => {
+    const currentListData = await currentList.get();
+    const itemList = currentListData.data();
+    if (itemList[itemName]) {
+      return true;
+    }
+    return false;
+  };
+
+  const updateFirestore = (currentList, itemName, formData) => {
     currentList
       .update({
-        items: firebase.firestore.FieldValue.arrayUnion({
-          name: data.itemName,
-          frequency: parseInt(data.itemFrequency),
+        [itemName]: {
+          name: formData.itemName,
+          frequency: parseInt(formData.itemFrequency),
           lastPurchased: null,
-        }),
+        },
       })
       .then(() => {
         alert('Item has been submitted!');
         reset();
+      })
+      .catch((e) => {
+        console.log(e);
+        alert('Item cannot be submitted at this time. Try again later.');
       });
   };
 
+  const onSubmit = async (data) => {
+    const currentList = db.collection('lists').doc(token);
```

Comment:
Change `'lists'` to something else to test the database error messages.

Reply:
This works but we are having a hard time understanding what this is for. A user isn't going to be able to change 'lists' to something else so if they see this, it's a dev's fault. Would a dev need more information on why this couldn't be submitted? Maybe add "The list isn't connected" or "There was an error with the database" or something similar. Or maybe direct it toward the user implying that this isn't their fault it's broken?

Reply:
Thanks for the feedback! Let's discuss the wording of the user-facing error messages on Merge Day so we can be on the same page. We've updated the console logs to signify which error the dev is encountering (eg. duplicate items error or updating Firestore error).

Reply:
Yes! This is great feedback, and I like how you're thinking about different levels and types of error messaging for the user and the dev side 👏